go/vt/vttablet/grpctmserver/server.go

```diff
@@ -200,6 +203,11 @@ func (s *server) UnlockTables(ctx context.Context, req *tabletmanagerdatapb.Unlo
 func (s *server) ExecuteQuery(ctx context.Context, request *tabletmanagerdatapb.ExecuteQueryRequest) (response *tabletmanagerdatapb.ExecuteQueryResponse, err error) {
 	defer s.tm.HandleRPCPanic(ctx, "ExecuteQuery", request, response, false /*verbose*/, &err)
 	ctx = callinfo.GRPCCallInfo(ctx)
+
+	// Attach the callerID as the EffectiveCallerID.
+	if request.CallerId != nil {
+		ctx = callerid.NewContext(ctx, request.CallerId, &querypb.VTGateCallerID{Username: request.CallerId.Principal})
+	}
```

Comment:
Note for future readers. We have not yet figured out whether ALL methods in this interface should process calledID in this way. In the meantime this PR fixes the known issue with `ApplySchema`.